back-end/app/Http/Controllers/Answer/WordsLearned.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace App\Http\Controllers\Answer;
+
+use App\Models\Lesson;
+use App\Models\Question;
+use Illuminate\Http\Request;
+use App\Models\Words_learned;
+use App\Http\Controllers\Controller;
+
+class WordsLearned extends Controller
+{
+    // Get words learned under lesson's taken with score and timestamps.
+    public function index()
+    {
+        $user = auth('sanctum')->user()->id;
+
+        // Filter the duplicated lesson id's.
+        $lessonTaken = Words_learned::where('user_id', $user)->get();
+
+        $lessonList = array();
+
+        foreach ($lessonTaken as $lesson) {
+            array_push($lessonList, $lesson->lesson_id);
+        }
+
+        $uniqueLessonList = array_keys(array_flip($lessonList));
+
+
+        // Get all the taken lesson of this user.
+        $getLessonDetails = array();
+
+        foreach ($uniqueLessonList as $lesson_id) {
+            array_push($getLessonDetails, $this->show($lesson_id)->original);
+        }
+
+        return response($getLessonDetails);
+    }
+
+    // Get words learned under lesson's taken with score and timestamps.
+    public function show($lesson_id)
+    {
+        $lesson = Words_learned::where('lesson_id', $lesson_id)->get();
+        $lessonTitle = Lesson::where('id', $lesson_id)->first()->title;
```

Comment:
```suggestion
        $lesson = Words_learned::with('lesson')->findOrFail($lessonId);
```

Reply:
Currently working on this,

Reply:
![image](https://user-images.githubusercontent.com/104751512/174548921-f13781be-390d-4181-8c3b-3270a32c64c8.png)

Sir Chev, I think I cant use with() method here, I'm using the $lesson to get all the words learned of a user in WordsLearned Table. That's why I used get() instead of findOrFail.

But I will consider using with() method in other controllers.

Reply:
![image](https://user-images.githubusercontent.com/104751512/174550395-3667c38b-e254-4250-8808-779134d7258a.png)

Sir Chev, I think I cant use with() method here, I'm using the $lesson to get all the words learned of a user in WordsLearned Table. That's why I used get() instead of findOrFail. I added user_id in where() method to count only the total score of that user in a particular lesson.

But I will consider using with() method in other controllers.